tests/unit/test_config.py

```diff
@@ -0,0 +1,72 @@
+import pytest
+from pydantic.error_wrappers import ValidationError
+
+from bugjira.config import Config
+
+
+def test_config_good_json(config_defaults):
+    """
+    GIVEN a json file containing a valid definition of a Bugjira config dict
+    WHEN we call Config.from_config with its pathname
+    THEN no exception is raised
+    """
+    Config.from_config(
+        config_path=(config_defaults + "/data/config/good_config.json")
+    )
+
+
+def test_config_good_dict(good_config_dict):
+    """
+    GIVEN a well-defined dict containing a Bugjira config
+    WHEN we call Config.from_config using the dict as the config_dict
+    THEN no exception is raised
+    AND the dict returned is the dict that was passed in
+    """
+    assert Config.from_config(config_dict=good_config_dict) == good_config_dict
+
+
+def test_config_missing_values(config_defaults):
+    """
+    GIVEN a json file containing an incomplete definition of a Bugjira config
+    WHEN we call Config.from_config with the path to the file
+    THEN a ValidationError is raised
+    """
+    with pytest.raises(ValidationError):
+        Config.from_config(
+            config_path=(config_defaults + "/data/config/missing_config.json")
+        )
+
+
+def test_config_no_file(config_defaults):
+    """
+    GIVEN a filepath pointing to a non-existent file
+    WHEN we call Config.from_config using the filepath as the config_path
+    THEN a FileNotFoundError is raised
+    """
+    with pytest.raises(FileNotFoundError):
+        Config.from_config(
+            config_path=(config_defaults + "/data/config/does_not_exist.json")
+        )
+
+
+def test_config_both_path_and_dict(good_config_file_path, good_config_dict):
+    """
+    GIVEN the get_config method, a path to a valid config file, and a dict that
```

Comment:
I don't think 'the get_config method' is needed in the GIVEN, since the fact that it will be called is expressed in the WHEN.